lib/Doctrine/ORM/Tools/Pagination/Paginator.php

```diff
@@ -116,7 +116,7 @@ public function getIterator()
         $offset = $this->query->getFirstResult();
         $length = $this->query->getMaxResults();
 
-        if ($this->fetchJoinCollection) {
+        if ($this->fetchJoinCollection && strpos($this->query->getDQL(), 'HAVING') === false) {
```

Comment:
`strpos()` is really to be avoided for this kind of tests: an AST check is needed